lib/src/main/java/uk/gov/di/ipv/cri/address/library/service/ConfigurationService.java

```diff
@@ -35,18 +42,29 @@ public long getAddressSessionTtl() {
                 .orElse(DEFAULT_SESSION_ADDRESS_TTL_IN_SECS);
     }
 
-    private String getParameterName(SSMParameterName parameterName) {
+    public String getParameterName(SSMParameterName parameterName) {
         return String.format("/%s/%s", parameterPrefix, parameterName.parameterName);
     }
 
     public enum SSMParameterName {
         ADDRESS_SESSION_TABLE_NAME("AddressSessionTableName"),
-        ADDRESS_SESSION_TTL("AddressSessionTtl");
+        ADDRESS_SESSION_TTL("AddressSessionTtl"),
+        OS_API_KEY("OrdinanceSurveyAPIKey");
 
         public final String parameterName;
 
         SSMParameterName(String parameterName) {
             this.parameterName = parameterName;
         }
     }
+
+    public String getOsApiKey() {
+
+        return secretsProvider.get(getParameterName(SSMParameterName.OS_API_KEY));
+    }
+
+    // This is exposed here so that we can unit test URL failures
+    public String getOsPostcodeAPIUrl() {
```

Comment:
can we retrieve this url from Parameter Store so we can sub in a mock if needed?

Reply:
This is already mocked in unit tests, but uses a constant as it is unlikely to change - would you prefer we made it pull from the parameter store with a default instead?

Reply:
We may introduce a stubbed Ordnance Survey in staging, so it would be good to have this url as a parameter

Reply:
Updated to use parameter store